src/desktop_notifier/common.py

```diff
@@ -245,7 +245,11 @@ class Notification:
     """A unique identifier for this notification. Generated automatically if not
     passed by the client."""
 
-    _buttons_dict: dict[str, Button] = field(default_factory=dict)
+    _buttons_dict: dict[str, Button] = field(default_factory=dict, init=False, repr=False, compare=False)
+
+    @property
+    def buttons_dict(self):
+        return self._buttons_dict
```

Comment:
Could you either return something immutable here or a copy?

Reply:
See #203, it's an `immutabledict` now.